code/graph-algorithms/cycle_undirected_graph/print_cycle_linked_components.cpp

```diff
@@ -0,0 +1,164 @@
+#include <bits/stdc++.h>
+
+using namespace std;
+
+class Graph
+{
+private:
+	int NumOfVertex;
+	vector<int> *Vertex;
+	int *Color;// 0 represents white , 1 represents gray and 2 represents black
```

Comment:
Could we use the vector for `color`, `parent`, `Timer1`, `Timer2` instead of pointer ?